src/persist-client/src/internal/state_versions.rs

```diff
@@ -273,6 +273,12 @@ impl StateVersions {
 
                 shard_metrics.set_since(new_state.since());
                 shard_metrics.set_upper(new_state.upper());
+                shard_metrics
```

Comment:
preference for pushing these into the `size_metrics` call to limit the number of times we iterate through all the batches to count things for metrics

Reply:
spine batches can't really be moved there easily, but I can move the rest